### PR TITLE
Rename files about error codes

### LINKED ISSUE #60017: Error code files (`diagnostics.rs`) reserve a valuable module name
... that can be used for diagnostics-related code in general, rather than for error code descriptions specifically.

Suggestion: rename the files with error codes (there are about 10 of them) to `error_codes.rs` to free the name `diagnostics` for something more useful.

Additionally: rename `librustc_resolve/error_reporting.rs` to `librustc_resolve/diagnostics.rs`.

### PR DESCRIPTION
**Pull request by JohnTitor**
fixes #60017 

This PR will be failed in tidy.

<details>
<summary>The log is here:</summary>

```
tidy check
tidy error: duplicate error code: 411
tidy error: Documents\GitHub\rust\src\librustc_resolve\diagnostics.rs:83:             __diagnostic_used!(E0411);
tidy error: Documents\GitHub\rust\src\librustc_resolve\diagnostics.rs:84:             err.code(DiagnosticId::Error("E0411".to_owned()));
tidy error: duplicate error code: 424
tidy error: Documents\GitHub\rust\src\librustc_resolve\diagnostics.rs:90:             debug!("smart_resolve_path_fragment: E0424, source={:?}", source);
tidy error: Documents\GitHub\rust\src\librustc_resolve\diagnostics.rs:92:             __diagnostic_used!(E0424);
tidy error: Documents\GitHub\rust\src\librustc_resolve\diagnostics.rs:93:             err.code(DiagnosticId::Error("E0424".to_owned()));
some tidy checks failed
```

</details>

I'd like to fix this but I don't know what to do.
I will work on later. Please let me know if you have any solutions.

r? @petrochenkov

### CODE REVIEW

**Comment by petrochenkov** [comment]
cc @rust-lang/wg-diagnostics @GuillaumeGomez

File: src/librustc_resolve/error_reporting.rs [hunk 1]
```diff
@@ -1,856 +0,0 @@
-use std::cmp::Reverse;
```
Comment (petrochenkov) [inline src/librustc_resolve/error_reporting.rs]: Could you move this renaming into a separate commit, so git doesn't interpret it a `+2,494 −2,494` diff.

**Comment by GuillaumeGomez** [comment]
Well, looks good to me (renaming to `error_codes` sounds good to me).

**Review by oli-obk — APPROVED** [review]

File: src/librustc_resolve/error_reporting.rs (hunk 1 above)
Comment (JohnTitor) [reply src/librustc_resolve/error_reporting.rs]: Okay, I separated commits.

File: src/librustc_resolve/lib.rs [hunk 2]
```diff
@@ -65,16 +65,16 @@ use std::mem::replace;
...
 use rustc_data_structures::sync::Lrc;
 
-use error_reporting::{find_span_of_binding_until_next_binding, extend_span_to_previous_binding};
+use diagnostics::{find_span_of_binding_until_next_binding, extend_span_to_previous_binding};
 use resolve_imports::{ImportDirective, ImportDirectiveSubclass, NameResolution, ImportResolver};
 use macros::{InvocationData, LegacyBinding, ParentScope};
 
 type Def = def::Def<NodeId>;
 
 // N.B., this module needs to be declared first so diagnostics are
```
Comment (JohnTitor) [inline src/librustc_resolve/lib.rs]: Should I replace this `diagnostics` with `error codes`?

**Comment by JohnTitor** [comment]
Thank you @davidtwco! It seems good, I changed file name in tidy.

**Review by estebank — APPROVED** [review]

**Comment by petrochenkov** [comment]
@JohnTitor 
Could you 1) update `src/doc/rustc-ux-guidelines.md` and 2) rename `libsyntax/diagnostic_list.rs` into `error_codes.rs` and remove it from tidy.

File: src/librustc_resolve/lib.rs (hunk 2 above)
Comment (petrochenkov) [reply src/librustc_resolve/lib.rs]: Not, this is the "good" diagnostics.rs.

Comment (JohnTitor) [reply src/librustc_resolve/lib.rs]: Okay.

**Comment by JohnTitor** [comment]
@petrochenkov I pushed some commits.
